suite-common/wallet-utils/src/reviewTransactionUtils.ts

```diff
@@ -0,0 +1,336 @@
+import { versionUtils } from '@trezor/utils';
+import {
+    Account,
+    FormState,
+    GeneralPrecomposedTransactionFinal,
+    ReviewOutputState,
+} from '@suite-common/wallet-types';
+import { StakeFormState, ReviewOutput } from '@suite-common/wallet-types';
+import { TrezorDevice } from '@suite-common/suite-types';
+
+import { getShortFingerprint, isCardanoTx } from './cardanoUtils';
+
+export const getTransactionReviewOutputState = (
+    index: number,
+    buttonRequestsCount: number,
+): ReviewOutputState => {
+    if (index === buttonRequestsCount - 1) return 'active';
+    if (index < buttonRequestsCount - 1) return 'success';
+
+    return undefined;
+};
+
+export const getIsUpdatedSendFlow = (device: TrezorDevice) => {
+    const firmwareVersion = getFirmwareVersion(device);
+
+    return versionUtils.isNewerOrEqual(firmwareVersion, '2.6.0');
+};
+
+export const getIsUpdatedEthereumSendFlow = (
+    device: TrezorDevice,
+    network: Account['networkType'],
+) => {
+    if (network !== 'ethereum') return false;
+
+    const firmwareVersion = getFirmwareVersion(device);
+
+    // publicly introduced in 2.6.3, versions 2.6.1 and 2.6.2 were internal
+    return versionUtils.isNewer(firmwareVersion, '2.6.0');
+};
+
+const getCardanoTokenBundle = (account: Account, output: CardanoOutput) => {
+    // Transforms cardano's tokenBundle into outputs, 1 output per one token
+    // since suite supports only 1 token per output it will return just one item
+    if (!output.tokenBundle || output.tokenBundle.length === 0 || 'addressParameters' in output)
+        return undefined;
+
+    if (account.tokens) {
+        return output.tokenBundle
+            .map(policyGroup =>
+                policyGroup.tokenAmounts.map(token => {
+                    const accountToken = account.tokens!.find(
+                        currentToken =>
+                            currentToken.contract ===
+                            `${policyGroup.policyId}${token.assetNameBytes}`,
+                    );
+                    if (!accountToken) return;
+
+                    const fingerprint = accountToken.name
+                        ? getShortFingerprint(accountToken.name)
+                        : undefined;
+
+                    return {
+                        type: 'cardano',
+                        contract: output.address,
+                        balance: token.amount,
+                        symbol: token.assetNameBytes
+                            ? Buffer.from(token.assetNameBytes, 'hex').toString('utf8')
+                            : fingerprint,
+                        decimals: accountToken.decimals,
+                    };
+                }),
+            )
+            .flat();
+    }
+};
+
+type ConstructOutputsParams = {
+    precomposedTx: GeneralPrecomposedTransactionFinal;
+    decreaseOutputId: number | undefined;
+    account: Account;
+    precomposedForm: FormState | StakeFormState;
+};
+
+const constructOldFlow = ({
```

Comment:
Can you please add comments to this old vs new flow? What does that even mean? 😆 Rename these please, if possible.